Framework/MDAlgorithms/src/MDNorm.cpp

```diff
@@ -365,6 +365,32 @@ std::map<std::string, std::string> MDNorm::validateInputs() {
       }
     }
   }
+  // validate accumulation workspaces, if provided
+  // this only checks that, if given, the two workspaces
+  // have the same dimensions.  We verify that binning parameters
+  // are the same when we execute the algorithm.
+  boost::shared_ptr<IMDHistoWorkspace> tempNormWS =
+      this->getProperty("TemporaryNormalizationWorkspace");
+  Mantid::API::IMDHistoWorkspace_sptr tempDataWS =
+      this->getProperty("TemporaryDataWorkspace");
+  if (tempNormWS || tempDataWS) {
+    if ((!tempNormWS) || (!tempDataWS)) {
+      errorMessage.emplace(
+          "AccumulationWorkspaces",
+          "Must provide either no accumulation workspaces or,"
+          "both TemporaryNormalizationWorkspaces and TemporaryDataWorkspace");
+      for (size_t i = 0; i < 3; i++) {
```

Comment:
Why only 3 dimensions? First check if the number of dimensions is the same, then do it for all dimensions.